client/core/core_test.go

```diff
@@ -2714,6 +2728,243 @@ func TestTradeTracking(t *testing.T) {
+				},
+				{
+					OrderID: immediateOrders.executed.ID().Bytes(),
+					Status:  uint8(order.OrderStatusBooked), // invalid, inactive orders should not be returned by DEX!
+				},
+			},
+			expectOrderStatuses: map[order.OrderID]order.OrderStatus{
+				standingOrders.epoch.ID():               order.OrderStatusBooked,   // epoch => booked
+				standingOrders.booked.ID():              order.OrderStatusBooked,   // should not change, cannot return to epoch
+				standingOrders.bookedPendingCancel.ID(): order.OrderStatusBooked,   // no status change
+				immediateOrders.epoch.ID():              order.OrderStatusEpoch,    // should not change, cannot be booked
+				immediateOrders.executed.ID():           order.OrderStatusExecuted, // should not change, inactive cannot become active
+			},
+		},
+		{
+			name: "active becomes inactive",
+			clientOrders: []*trackedTrade{
+				standingOrders.epoch,
+				standingOrders.booked,
+				standingOrders.bookedPendingCancel,
+				standingOrders.executed,
+				immediateOrders.epoch,
+				immediateOrders.executed,
+			},
+			serverOrders: []*msgjson.Order{}, // no active order reported by server
+			orderStatuses: []msgjson.OrderStatusResult{
+				{
+					OrderID: standingOrders.epoch.ID().Bytes(),
+					Status:  uint16(order.OrderStatusRevoked),
+				},
+				{
+					OrderID: standingOrders.booked.ID().Bytes(),
+					Status:  uint16(order.OrderStatusRevoked),
+				},
+				{
+					OrderID: standingOrders.bookedPendingCancel.ID().Bytes(),
+					Status:  uint16(order.OrderStatusCanceled),
+				},
+				{
+					OrderID: immediateOrders.epoch.ID().Bytes(),
+					Status:  uint16(order.OrderStatusExecuted),
+				},
+			},
+			expectOrderStatuses: map[order.OrderID]order.OrderStatus{
+				standingOrders.epoch.ID():               order.OrderStatusRevoked,  // preimage missed = revoked
+				standingOrders.booked.ID():              order.OrderStatusRevoked,  // booked, not canceled = assume revoked (may actually be executed)
+				standingOrders.bookedPendingCancel.ID(): order.OrderStatusCanceled, // booked pending canceled = assume canceled (may actually be revoked or executed)
+				standingOrders.executed.ID():            order.OrderStatusExecuted, // should not change
+				immediateOrders.epoch.ID():              order.OrderStatusExecuted, // preimage sent, not canceled = executed
+				immediateOrders.executed.ID():           order.OrderStatusExecuted, // should not change
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		// Track client orders in dc.trades.
+		dc.tradeMtx.Lock()
+		var pendingCancel *trackedTrade
+		dc.trades = make(map[order.OrderID]*trackedTrade)
+		for _, tracker := range tt.clientOrders {
+			dc.trades[tracker.ID()] = tracker
+			if tracker.cancel != nil {
+				pendingCancel = tracker
+			}
+		}
+		dc.tradeMtx.Unlock()
+
+		if pendingCancel != nil {
+			// Canceling the order again should fail.
+			rig.queueCancel(nil)
```

Comment:
The cancel route should not get hit, so this is at best unnecessary, at worst, potentially problematic.

Reply:
Allowing this queued response to ensure that the error returned here is the expected error and not a `response not queued for route` error. The queued response should actually only be served later in this tests and will error then if it was already served here.